### PR TITLE
jssdk: refetch config when iframe emits message

### PR DESCRIPTION
**Pull request by chantalchung**
- listen for `DVC.optIn.saved` event message

### CODE REVIEW

File: sdk/js/src/Client.ts [hunk 1]
```diff
@@ -75,6 +75,14 @@ export class DVCClient implements Client {
                 this.eventEmitter.emitError(err)
                 return this
             })
+            
+        if (!options?.reactNative && typeof window !== 'undefined') {
+            window.addEventListener('message', (event) => {
+                if (event.data === 'DVC.optIn.saved') {
```
Comment (laurawarr) [inline sdk/js/src/Client.ts:81]: should we update this and the opt-in widget to use the object structure?
```suggestion
                const message = event.data
                if (message?.type === 'DVC.optIn.saved') {
```

Comment (chantalchung) [reply sdk/js/src/Client.ts:81]: yup, will make a PR to update the message it in the widget

**Review by laurawarr — APPROVED** [review]

File: sdk/js/src/Client.ts [hunk 2]
```diff
@@ -75,6 +75,15 @@ export class DVCClient implements Client {
                 this.eventEmitter.emitError(err)
                 return this
             })
+            
+        if (!options?.reactNative && typeof window !== 'undefined') {
+            window.addEventListener('message', (event) => {
```
Comment (ajwootto) [inline sdk/js/src/Client.ts:80]: Can you also unregister this listener in the `close` method?

Comment (chantalchung) [reply sdk/js/src/Client.ts:80]: sorry, I'll make a PR to add this!

**Pull request merged**